pkg/spireentry/reconciler.go

```diff
@@ -358,6 +359,17 @@ func (r *entryReconciler) addClusterStaticEntryEntriesState(ctx context.Context,
 
 func (r *entryReconciler) addClusterSPIFFEIDEntriesState(ctx context.Context, state entriesState, clusterSPIFFEIDs []*ClusterSPIFFEID) {
 	log := log.FromContext(ctx)
+	seen := make(map[string]map[string]bool)
```

Comment:
I think the code can be simplified if we track by pod UID instead of using nested maps:
```suggestion
	podsWithNonFallbackApplied := make(map[string]struct{})
```
And then below:
```
					if !clusterSPIFFEID.Spec.Fallback {
						podsWithNonFallbackApplied[pods[i].UID] = struct{}{}
					}
```